client/state/current-user/actions.js

```diff
@@ -23,3 +35,39 @@ export function setCurrentUserFlags( flags ) {
 		flags
 	};
 }
+
+export function uploadGravatar( file, bearerToken, email ) {
+	return dispatch => {
+		dispatch( { type: GRAVATAR_UPLOAD_START } );
+		const data = new FormData();
+		data.append( 'filedata', file );
+		data.append( 'account', email );
+		return request
+			.post( 'https://api.gravatar.com/v1/upload-image' )
+			.send( data )
+			.set( 'Authorization', 'Bearer ' + bearerToken )
+			.set( 'Accept-Language', '*' )
+			.then( () => {
+				dispatch( successNotice(
```

Comment:
There's also the [notices middleware](https://github.com/Automattic/wp-calypso/blob/master/client/state/notices/middleware.js) for dispatching notices in response to actions. Having access to global state there has the advantage too of allowing the notice to only be shown in particular sections or contexts, for example. See also: #6645